### PR TITLE
Make schema loader beta

### PR DESCRIPTION
**Pull request by ColinKerr**
ecschema-metadata
- Made SchemaLoader, SchemaJsonLocater and SchemaPropsGetter beta
- Add getter to SchemaLoader for context

ecschema-locaters
- Change SchemaFileUtility class to SchemaXml namespace and make beta
- Rename functions to be shorter and clearer

Added docs for loading schemas from an iModel, creating a new schema with the editor and inserting back into the iModel
Added docs for creating and using code snippets

### CODE REVIEW

File: common/api/ecschema-locaters.api.md [hunk 1]
```diff
@@ -43,6 +43,12 @@ export abstract class SchemaFileLocater {
     searchPaths: string[];
 }
 
+// @beta
+export class SchemaFileUtility {
```
Comment (pmconne) [inline common/api/ecschema-locaters.api.md:47]: This looks like it's supposed to be a `namespace`, not a `class`. The word "utility" in a type name usually indicates the type is just a grab-bag container. Consider just exporting two free functions or, if you must keep them in a container object, give the container a better name.

**Review by pmconne — COMMENTED** [review]
Please summarize these changes and promotions with links to documentation in NextVersion.md.

File: common/api/ecschema-locaters.api.md (hunk 1 above)
Comment (ColinKerr) [reply common/api/ecschema-locaters.api.md]: done

**Review by pmconne — APPROVED** [review]

**Comment by ColinKerr** [comment]
> Please summarize these changes and promotions with links to documentation in NextVersion.md.

done

File: docs/learning/serializing-xml-schemas.md [hunk 2]
```diff
@@ -6,14 +6,14 @@ Bentley provided package [@itwin/ecschema-locaters](https://www.itwinjs.org/refe
...
 
-## SchemaFileUtility in @itwin/ecschema-locaters
+## SchemaXml in @itwin/ecschema-locaters
 
-The SchemaFileUtility class in the [@itwin/ecschema-locaters](https://www.itwinjs.org/reference/ecschema-locaters) package
-contains the 'writeSchemaXmlFile' method. This method takes the EC Schema instance and a path to the file to create. The
-SchemaFileUtility.writeSchemaXmlFile method is asynchronous, requiring the 'await' keyword.
+The SchemaXml class in the [@itwin/ecschema-locaters](https://www.itwinjs.org/reference/ecschema-locaters) package
+contains the 'writeFile' method. This method takes the EC Schema instance and a path to the file to create. The
+SchemaXml.writeFile method is asynchronous, requiring the 'await' keyword.
```
Comment (pmconne) [inline docs/learning/serializing-xml-schemas.md:13]:
```suggestion
The [SchemaXml.writeFile]($ecschema-locaters) method takes the EC Schema instance and serializes it to a specified location on disk. Note: this method is `async`, requiring the caller to `await` it.
```
doc links, backticks.

File: docs/learning/serializing-xml-schemas.md [hunk 3]
```diff
@@ -49,3 +49,23 @@ We can now use the file system API to write the xml to file.
 ```ts
 [[include:Serialize_Schema_To_XML_Write]]
 ```
+
+## Schemas from an iModel
+
+First use the SchemaLoader from the [@itwin/ecschema-metadata](https://www.itwinjs.org/reference/ecschema-metadata/) package to create a SchemaContextEditor that knows about the schemas in the iModel
```
Comment (pmconne) [inline docs/learning/serializing-xml-schemas.md:55]: `SchemaContextEditor` is `alpha` - it shouldn't appear in documentation or example code.

Comment (ColinKerr) [reply docs/learning/serializing-xml-schemas.md]: It was already used in the samples and we are finally getting started working on it again and getting some people to prototype with it so I'd like to leave the references in there with the knowledge that we will probably bring it up to beta once we have a bit more work done and a bit more feedback on it

Comment (pmconne) [reply docs/learning/serializing-xml-schemas.md]: Please promote it ASAP, broken links in docs are bad.

File: example-code/snippets/src/backend/SerializingXmlSchemas.ts [hunk 4]
```diff
@@ -14,9 +14,9 @@ export class SchemaXmlSerializer {
   public async serializeSchemaToXmlSample() {
```
Comment (christophermlawson) [inline example-code/snippets/src/backend/SerializingXmlSchemas.ts:14]: @ColinKerr Should we also reference SchemaXml.writeFile function from ecschema-locaters as an alternative to this?

Comment (ColinKerr) [reply example-code/snippets/src/backend/SerializingXmlSchemas.ts:14]: This is the documentation for if you do not want to use those helpers.  See serializing-xml-schemas.md for the docs that reference this

Comment (christophermlawson) [reply example-code/snippets/src/backend/SerializingXmlSchemas.ts:14]: Got it.  Thanks.

**Review by christophermlawson — APPROVED** [review]

**Pull request merged**